crates/wasm-wave/src/lex.rs

```diff
@@ -0,0 +1,137 @@
+//! Lexing types
+
+use std::fmt::Display;
+
+pub use logos::Span;
+
+/// A WAVE `logos::Lexer`
+pub type Lexer<'source> = logos::Lexer<'source, Token>;
+
+/// A WAVE token
+#[derive(Clone, Copy, Debug, PartialEq, Eq, logos::Logos)]
+#[logos(error = Option<Span>)]
+#[logos(skip r"[ \t\n\r]+")]
+#[logos(skip r"//[^\n]*")]
+#[logos(subpattern label_word = r"[a-z][a-z0-9]*|[A-Z][A-Z0-9]*")]
+#[logos(subpattern char_escape = r#"\\['"tnr\\]|\\u\{[0-9a-fA-F]{1,6}\}"#)]
```

Comment:
@lann I'm not familiar with logos but how difficult do you think it would be to hand-writet he lexer instead of using logos? I suspec that's a relatively big dep for Wamtime itself and while it's not the worst thing in the world to audit it might also be reasonable to hand-roll a lexer for this (there's also some precedent give the... two? ... other hand-rolled lexers in this repo)

If it's difficult to rewrite or logos gives some really nice features then I think it's fine to keep logos, though.

Reply:
Wave started with a hand-rolled lexer. I only switched because of a plan to embed in WAC's grammer which uses Logos.

cc @fibonacci1729

Reply:
Ah ok makes sense! I that case I think it's reasonable to keep this. I should investigate and perahps we should change all the other lexers in this repository to also use logos...

Reply:
Logos is quite nice; especially when wanting to compose lexers across different crates.